bcda/client/bluebutton.go

```diff
@@ -103,10 +103,16 @@ func (bbc *BlueButtonClient) GetPatient(patientID, jobID, cmsID string) (string,
 	return bbc.getData(blueButtonBasePath+"/Patient/", params, jobID, cmsID)
 }
 
-func (bbc *BlueButtonClient) GetPatientByHICNHash(hashedHICN string) (string, error) {
+func (bbc *BlueButtonClient) GetPatientByIdentifierHash(hashedIdentifier, patientIdMode string) (string, error) {
 	params := GetDefaultParams()
+
+	identifier := "hicnHash"
+	if patientIdMode == "MBI_MODE" {
+		identifier = "mbiHash"
+	}
+
 	// FHIR spec requires a FULLY qualified namespace so this is in fact the argument, not a URL
-	params.Set("identifier", fmt.Sprintf("http://bluebutton.cms.hhs.gov/identifier#hicnHash|%v", hashedHICN))
+	params.Set("identifier", fmt.Sprintf("http://bluebutton.cms.hhs.gov/identifier#%s|%v", identifier, hashedIdentifier))
```

Comment:
This should change to `https://bluebutton.cms.gov/resources/identifier/%s|%v` according to: https://github.com/CMSgov/beneficiary-fhir-data/blob/0877232340983fda8dbcd202012e991b8b531ac5/apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerConstants.java#L78

Reply:
done will be in next commit